src/interfaces/DiscreteGenerator.ts

```diff
@@ -64,4 +65,11 @@ export interface DiscreteGenerator<TShared, TInternal> {
      * @param newShared The parameter set to update from
      */
     updateShared(shared: TShared, newShared: TShared): void;
+
+    /**
+     * Gets a {@link ZeroEvery} vector used to reset certain indices of state to 0 at a
+     * given frequency
+     * @param shared The shared parameter set to update
+     */
+    zeroEvery?(shared: TShared): ZeroEvery;
```

Comment:
The idea with including shared here is that the parameter values themselves might define the zeroing behaviour, rather than it just being something fixed like "incidence gets zeroed every day for all parameter values"...?